[GTK][WPE] Use g_error() instead of _exit() when web process is forced to exit after connection is closed

carlosgcampos opened the pull request:
#### f748569093942a2282672ae5e0da6eefaf45bd69
<pre>
[GTK][WPE] Use g_error() instead of _exit() when web process is forced to exit after connection is closed
<a href="https://bugs.webkit.org/show_bug.cgi?id=250377">https://bugs.webkit.org/show_bug.cgi?id=250377</a>

Reviewed by Michael Catanzaro.

To make it easier to debug the cause of the process leak.

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::crashAfter10Seconds):
(WebKit::WebProcess::initializeConnection):
(WebKit::callExitSoon): Deleted.

Canonical link: <a href="https://commits.webkit.org/258823@main">https://commits.webkit.org/258823@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25a412cb36e0869b426167fedadd733232508eae

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102928 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12048 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35952 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112178 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2954 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95167 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109833 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108702 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10039 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93229 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37669 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91881 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24763 "Found 2 new test failures: http/tests/security/clean-origin-css-exposed-resource-timing.html, http/tests/security/cross-origin-clean-css-resource-timing.html (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79405 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5488 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26175 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5641 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2627 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11653 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45671 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7389 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3229 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -266,7 +266,10 @@ using namespace WebCore;
-#if OS(WINDOWS)
+#if PLATFORM(GTK) || PLATFORM(WPE)
+    // We use g_error() here to cause a crash and allow debugging this unexpected late exit.
+    g_error("WebProcess didn't exit as expected after the UI process connection was closed");
```
The only problem here is nomenclature: the name of this function is `callExit` but now it's calling something else instead. I would do this:

 * Add `#if !PLATFORM(GTK) && !PLATFORM(WPE)` guards around `callExit`
 * Call `g_error` directly in `callExitSoon` instead of using `callExit`
 * Rename `callExitSoon` to `crashAfter10Seconds`

mcatanzaro reviewed (commented):
Thank you! One more fewer thing for me to worry about....

carlosgcampos committed:
[GTK][WPE] Use g_error() instead of _exit() when web process is forced to exit after connection is closed
https://bugs.webkit.org/show_bug.cgi?id=250377

Reviewed by Michael Catanzaro.

To make it easier to debug the cause of the process leak.

* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::crashAfter10Seconds):
(WebKit::WebProcess::initializeConnection):
(WebKit::callExitSoon): Deleted.

Canonical link: https://commits.webkit.org/258823@main
